### DISCUSSION TITLE
How to package a multi-entry application into an appimage and use it?

### DISCUSSION THREAD

**Discussion by masx200** [AppImage/AppImageKit#1250] [Q&A]
How to package a multi-entry application into an appimage?

For example, openjdk, there are many executable programs, how to write apprun files? How to use the packaged multi-entry appimage?

**Comment by probonopd** [comment]
AppImage is designed to run one GUI application. Say, you have an IDE that uses openjdk, then double-clicking the AppImage should open the IDE.

**Comment by masx200** [comment]
For programs like openjdk, should the correct way to use it be set to automatically mount and add environment variables at boot time?

```
openjdk.appimage --appimage-mount
```

**Comment by probonopd** [comment]
The way I see it, OpenJDK itself is not an application. The application that needs OpenJDK should include it in its AppImage.

Would that work for your use case? What do you want to achieve in the end?

**Comment by TheAssassin** [comment]
@probonopd please stop replying in issues that are questions before converting them to discussions.

**Reply by probonopd** [reply]
The comments travel with the move, so I never thought about whether the order of doing these things makes any difference. Apparently it does?
